test/jdk/java/nio/channels/FileChannel/Transfer4GBFile.java

```diff
@@ -51,72 +50,53 @@ public class Transfer4GBFile {
     // Test transferTo with large file
     @Test
     public void xferTest04() throws Exception { // for bug 4638365
-        File source = File.createTempFile("blah", null);
-        source.deleteOnExit();
+        Path source = FileUtils.createSparseTempFile("blah", null);
+        source.toFile().deleteOnExit();
         long testSize = ((long)Integer.MAX_VALUE) * 2;
-        initTestFile(source, 10);
-        RandomAccessFile raf = new RandomAccessFile(source, "rw");
-        FileChannel fc = raf.getChannel();
+
         out.println("  Writing large file...");
         long t0 = System.nanoTime();
-        fc.write(ByteBuffer.wrap("Use the source!".getBytes()), testSize - 40);
-        long t1 = System.nanoTime();
-        out.printf("  Wrote large file in %d ns (%d ms) %n",
-            t1 - t0, TimeUnit.NANOSECONDS.toMillis(t1 - t0));
-
-        fc.close();
-        raf.close();
-
-        File sink = File.createTempFile("sink", null);
-        sink.deleteOnExit();
+        try (FileChannel fc = FileChannel.open(source,
+                StandardOpenOption.READ, StandardOpenOption.WRITE)) {
+            fc.write(ByteBuffer.wrap("Use the source!".getBytes()), testSize - 40);
+            long t1 = System.nanoTime();
+            out.printf("  Wrote large file in %d ns (%d ms) %n",
+                    t1 - t0, TimeUnit.NANOSECONDS.toMillis(t1 - t0));
+        }
 
-        FileInputStream fis = new FileInputStream(source);
-        FileChannel sourceChannel = fis.getChannel();
+        Path sink = Files.createTempFile("sink", null);
+        sink.toFile().deleteOnExit();
 
-        raf = new RandomAccessFile(sink, "rw");
-        FileChannel sinkChannel = raf.getChannel();
+        try (FileChannel sourceChannel = FileChannel.open(source,
+                     StandardOpenOption.READ);
+             FileChannel sinkChannel = FileChannel.open(sink,
+                     StandardOpenOption.WRITE)) {
```

Comment:
Some weird indentation at lines 71 and 73. A static import of the `StandardOpenOption` enum onstants would help.